src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/LanguageServices/LanguageServiceHost.cs

```diff
@@ -113,9 +117,6 @@ private Task OnProjectFactoryCompletedAsync()
 
         private async Task OnProjectChangedCoreAsync(IProjectVersionedValue<IProjectSubscriptionUpdate> e, RuleHandlerType handlerType)
         {
-            // TODO: https://github.com/dotnet/roslyn-project-system/issues/353
-            await _commonServices.ThreadingService.SwitchToUIThread();
```

Comment:
I moved this call inside the HandleAsync as we now execute the core handler callback inside joinable tasks, and that needs to force UI thread anyways.